src/computational_graph/diagram.jl

```diff
@@ -169,12 +222,62 @@ function Base.:+(g1::Diagram{F,W}, g2::Diagram{F,W}) where {F,W}
     type = g1.type
     @assert Set(g1.external_vertices) == Set(g2.external_vertices)
     #TODO: add external vertices creation/annihilation check
-    return Diagram{F,W}(g1.external_vertices; type=type, is_connected=g1.is_connected && g2.is_connected, subdiagram=[g1, g2], operator=Sum())
+    return Diagram{F,W}(g1.external_vertices; type=type, is_connected=g1.is_connected && g2.is_connected, couplings=g1.couplings, subdiagram=[g1, g2], operator=Sum())
 end
 
 function Base.:-(g1::Diagram{F,W}, g2::Diagram{F,W}) where {F,W}
     return g1 + (-1) * g2
 end
+
+"""
+    function ExternalVertices(V1::Vector{ExternalVertex}, V2::Vector{ExternalVertex}, couplings::Couplings)
+
+    Give external vertices when two diagrams combine.
+    
+    # Arguments
+    - V1::Vector{ExternalVertex}     : External vertices of diagram I.
+    - V2::Vector{ExternalVertex}     : External vertices of diagram II.
+    - couplings::Couplings           : all the vertex couplings in the Feynman rule. 
+"""
+function ExternalVertices(V1::Vector{ExternalVertex}, V2::Vector{ExternalVertex}, couplings::Couplings)
```

Comment:
I would name this function `getExternalVertices`. To me, `ExternalVertices` would be a `Vector{ExternalVertex}`.